packages/flutter/pubspec.yaml

```diff
@@ -15,6 +15,7 @@ dependencies:
   vector_math: 2.1.0
   sky_engine:
     sdk: flutter
+  intl: ^0.17.0
```

Comment:
Adding this dependency is not a good idea. This will force all users of flutter to use a version of package:intl that's compatible with this version. That's something we are trying to avoid for any package.

Reply:
Are you suggesting that I downgrade the version of intl that is being imported? Or do I attempt to reimplement patterns from intl without creating a dependency in order to support intl functionality?? What is the appropriate way to support intl?

Reply:
I'm not sure what exactly @goderbauer meant (he'll be out of office for a few weeks), but I think he means that we try to avoid adding the package since it means that users will then be forced to use an `intl` version that is not only compatible with `flutter` but also with their own Flutter app?

Reimplementing patterns for `DateFormat` could also be problematic if we called the class `DateFormat` because anybody who _is_ using the `intl` package would have naming conflicts unless they used an alias for the `intl` package from the get-go.

The only solution I can think of is to then reimplement the `DateFormat` parsing, but call it something else, but that sounds more confusing than helpful for anybody familiar with `intl`'s `DateFormat`.

Reply:
Maybe a `DateFormatStrategy` class with a callback (or just a callback) that we can use to format the date. Here is an example:

```dart
class DateFormatStrategy {
  DateFormatStrategy._(this.formattedDate);
  
  final String formattedDate;
  
  factory DateFormatStrategy.parse(String formattedDate) {
    return DateFormatStrategy._(formattedDate);
  }
}
```

Then, we could use that class instead of `DateFormat`.

Reply:
@goderbauer @darrenaustin do you think that the suggestion in my comment above would be a valid solution for this?